src/tests/LearnositySdk/Request/InitTest.php

```diff
@@ -67,12 +67,42 @@ public function testGenerateSignature($expectedResult, $initObject)
      */
     public function testGenerate($expectedResult, $initObject)
     {
+        $generatedWasString = false;
+        $requestWasString = false;
         $generated = $initObject->generate();
 
         if (is_array($expectedResult)) {
             ksort($expectedResult);
         }
 
+        if (is_string($generated)) {
```

Comment:
Yeah, ok, that's pretty ugly. We may have a problem there \:

Reply:
@shtrom ^^^ what problem? it's ugly I agree, but `meta` must be removed as it may be different on each platform running unit tests.

Reply:
Yeah, I agree. I was following up on your previous comment that we may have a problem. We can't reasonably require to do this sort of jumping around to work...

That said, I understand your initial question better, and all of this maybe due to my misunderstanding. When you asked whether we should include the `meta` in the signature, I thought you meant adding the key specifically as some field we want to sign, but I know think that you meant “Shall we leave it in the full request that gets appended at the end of the signature string when the API requires it?” I think this could actually work, and this would save us from doing all this.

However, we'll have to update the signatures for all the tests, which will break reference tests across SDK codebase.

At this stage, I'd be tempted to do the following:
1. Include (i.e., don't remove) the `meta` in the full `requestString` that gets appended to the string to be signed
2. Add a method to `Init` which allows to _disable_ the addition of the `meta` (i.e., probably just setting a `$this->addMeta = false`), so we can test the signature-generation on deterministic payloads across the codebases.

Do you think it would make sense?

Reply:
@shtrom yep this suggestion makes sense.

Although it still won't solve our "different floating-point when decoding/encoding" issue...

Reply:
Signature generation is done in the constructor therefore I decided to add an extra parameter to it. This creates a problem though - this parameter can easily be used by clients as well to not append `meta` as it's not "hidden enough".

But this parameter may be used by clients if they hit our "different floating-point when decoding/encoding" issue, so I think this may be a good solution. I've added test to cover this case specifically as well.

So I think after weighing all in, it's a good solution.

Reply:
Hum, I would avoid adding anything to the constructor, as it can get confusing and unwieldy. I would prefer it to be a separate accessor altogether.